frontend/packages/pipelines-plugin/src/components/pipelines/pipeline-builder/update-utils.ts

```diff
@@ -434,26 +486,40 @@ const fixInvalidListTask: UpdateOperationAction<UpdateOperationFixInvalidTaskLis
   };
 };
 
+const addFinallyListTask: UpdateOperationAction<UpdateOperationFixInvalidTaskListData> = (
+  taskGroup,
+) => {
+  const newTaskName = `finally-list-${getRandomChars(6)}`;
+  const { finallyListTasks } = taskGroup;
+  return {
+    ...taskGroup,
+    finallyListTasks: [...finallyListTasks, { name: newTaskName }],
+  };
+};
+
 export const applyChange = (
   taskGroup: PipelineBuilderTaskGroup,
   op: UpdateOperation,
 ): CleanupResults => {
   const { type, data } = op;
-  const { tasks, listTasks } = taskGroup;
```

Comment:
You went a different direction, so let us make it more pure than this currently.

```
const { tasks, listTasks, finallyTasks, finallyListTasks } = taskGroup;
const taskGrouping: PipelineBuilderTaskGrouping = { tasks, listTasks, finallyTasks, finallyListTasks };
// and pass `taskGrouping` into all the methods.
```